internal/storage/bucket_handle.go

```diff
@@ -280,6 +274,14 @@ func (b *bucketHandle) ListObjects(ctx context.Context, req *gcs.ListObjectsRequ
 			currObject := storageutil.ObjectAttrsToBucketObject(attrs)
 			list.Objects = append(list.Objects, currObject)
 		}
+
+		// itr.next returns all the objects present in the bucket. Hence adding a
+		// check to break after required number of objects are returned.
```

Comment:
We can change the comment here -

"after required number of objects are returned" ->  something like "after iterating over the current page"